lib/chainlock/chainlock.js

```diff
@@ -22,310 +22,310 @@ const bls = require('../crypto/bls');
  * @property {Buffer} blockHash
  * @property {Buffer} signature
  */
-function ChainLock(arg) {
-  if (arg instanceof ChainLock) {
-    return arg.copy();
+class ChainLock {
+  constructor(arg) {
```

Comment:
I would accept only raw info (as plain object) to constructor as we are doing in DPP and if you need to create chainlock from something else you should use `from` (such as `Buffer`) or specific `from[Something]`

Reply:
@Alex-Werner @antouhou You thoughts? I know that the rest of the library accepts any shit to constructor but probably it's time to improve library API step by step?

Reply:
In term of developer experience, the new ChainLock(whatever) is the whole difference between having to use the documentation or not. 
I'm actually in favour of detecting input type, parsing it to output a object and use that object to init the constructor for the user. 
One issue is in the way we actually perform that in the API imho, it's ugly, but the dev xperience is clearly there :D

Reply:
I tend to agree with Alex. The example you mention @shumkov will actually be the main use case for light client implementation as they will get the chainlocks as a buffer from the stream. Not having to write ChainLock.fromBuffer(buf) seems to worth it to bear the uglyness. Wonder though what @antouhou  thinks.

Reply:
I'll probably agree with @Alex-Werner and @Cofresi , but for a different reason - consistency with how the rest of the classes in the lib is used

Reply:
Ok... But I meant the consistency with other code what we have: DPP, Drive, and so on. The dashcore-lib code is not great we all know. So it's just a question: do we want to keep new code in this library consistent with other our code or just consistent with the old code of the library.

Reply:
I would keep the interfaces/constructors consistent with the old code first. That way we can incrementally update the old code to ES6+ without worrying about having to change other libraries. Then if there is a need and a will we can change constructors in a second overhaul step?

Reply:
Make sense!